pygeo/constraints/DVCon.py

```diff
@@ -3204,12 +3221,56 @@ def _generateIntersections(self, leList, teList, nSpan, nChord, surfaceName):
         root_s = Curve(X=[leList[0], teList[0]], k=2)
         tip_s = Curve(X=[leList[-1], teList[-1]], k=2)
 
-        # Generate parametric distances
-        span_s = np.linspace(0.0, 1.0, nSpan)
+        # Generate spanwise parametric distances
+        if isinstance(nSpan, int):
+            # Use equal spacing along the curve
+            le_span_s = te_span_s = np.linspace(0.0, 1.0, nSpan)
+        elif isinstance(nSpan, list):
+            # Use equal spacing within each segment defined by leList and teList
+
+            # We use the same nSpan for the leading and trailing edges, so check that the lists are the same size
+            if len(leList) != len(teList):
+                raise ValueError("leList and teList must be the same length if nSpan is provided as a list.")
```

Comment:
I know it's mentioned in the docstring above, but what are the instances where `leList` and `teList` would be intentionally inconsistent when `nSpan` is float?

Reply:
That's a good question. I can't see it happening too often, but there could be some uses. For example, a wing could have a leading edge break and a straight trailing edge.

Reply:
Oh yeah, that makes sense.